service/dap/server.go

```diff
@@ -1730,12 +1730,23 @@ func (s *Server) onEvaluateRequest(request *dap.EvaluateRequest) {
 			s.sendErrorResponseWithOpts(request.Request, UnableToEvaluateExpression, "Unable to evaluate expression", err.Error(), showErrorToUser)
 			return
 		}
+
+		// The return values of injected function calls are volatile.
+		// Load as much useful data as possible.
+		// - String: a common use case of call injection is to stringify complex
+		// data conveniently. That can easily exceed the default limit, 64.
```

Comment:
IMO, it's better to relax the string type variable length limit for non-call evaluate requests sent with watch context too (as long as MaxVariableRecurse remains 1 and we relax the limit only on watch mode).

Watch is more like a short cut to automate the evaluation requests that users would've run through DEBUG CONSOLE if Watch wasn't available. It will be more convenient if they can see the full values without copying or running them again from DEBUG CONSOLE. Users can remove/edit them any time (unlike those in Variables section) if they find they are loading too much and slowing down the overall performance.

1K doesn't seem a lot, but one thing that we still need to address is to limit the size of the value string for compound types (https://github.com/golang/vscode-go/issues/1435). For map or struct with many long strings, I am afraid this can be huge. I think we can investigate it in a separate PR to address the issue and try to truncate the final value string.